masonry/src/tree_arena.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO - Factor out into a separate crate.
+#![allow(dead_code)]
+
+struct TreeNode<Item> {
+    id: u64,
+    item: Item,
+    children: Vec<TreeNode<Item>>,
+}
+
+#[derive(Default)]
+pub struct TreeArena<Item> {
```

Comment:
Every `pub` in this module should be `pub(crate)`

Reply:
I get your point, but I think the "this will factored out" aspect is clearer with pub items.